src/ParticleContainer.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * ParticleContainer.h
+ */
+
+#pragma once
+
+#include "Particle.h"
+#include <array>
+#include <memory>
+#include <stdexcept>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+using ParticlePointer = std::shared_ptr<Particle>;
+
+struct ParticlePair {
```

Comment:
doxygen comments missing for the struct and all variables, methods inside! it also makes it quite hard to understand why do you have some of the methods here.